src/System.Net.HttpListener/src/System/Net/Unix/HttpConnection.cs

```diff
@@ -0,0 +1,552 @@
+
+        public HttpRequestStream GetRequestStream(bool chunked, long contentlength)
+        {
+            if (_requestStream == null)
+            {
+                byte[] buffer = _memoryStream.GetBuffer();
+                int length = (int)_memoryStream.Length;
+                _memoryStream = null;
+                if (chunked)
+                {
+                    _chunked = true;
+                    _context.Response.SendChunked = true;
+                    _requestStream = new ChunkedInputStream(_context, _stream, buffer, _position, length - _position);
+                }
+                else
+                {
+                    _requestStream = new HttpRequestStream(_stream, buffer, _position, length - _position, contentlength);
+                }
+            }
+            return _requestStream;
+        }
+
+        public HttpResponseStream GetResponseStream()
+        {
+            if (_responseStream == null)
+            {
+                HttpListener listener = _context.Listener;
+
+                if (listener == null)
+                    return new HttpResponseStream(_stream, _context.Response, true);
+
+                _responseStream = new HttpResponseStream(_stream, _context.Response, listener.IgnoreWriteExceptions);
+            }
+            return _responseStream;
+        }
+
+        private static void OnRead(IAsyncResult ares)
+        {
+            HttpConnection cnc = (HttpConnection)ares.AsyncState;
+            cnc.OnReadInternal(ares);
+        }
+
+        private void OnReadInternal(IAsyncResult ares)
+        {
+            _timer.Change(Timeout.Infinite, Timeout.Infinite);
+            int nread = -1;
+            try
+            {
+                nread = _stream.EndRead(ares);
+                _memoryStream.Write(_buffer, 0, nread);
+                if (_memoryStream.Length > 32768)
+                {
+                    SendError(HttpStatusDescription.Get(400), 400);
+                    Close(true);
+                    return;
+                }
+            }
+            catch
+            {
+                if (_memoryStream != null && _memoryStream.Length > 0)
+                    SendError();
+                if (_socket != null)
+                {
+                    CloseSocket();
+                    Unbind();
+                }
+                return;
+            }
+
+            if (nread == 0)
+            {
+                CloseSocket();
+                Unbind();
+                return;
+            }
+
+            if (ProcessInput(_memoryStream))
+            {
+                if (!_context.HaveError)
+                    _context.Request.FinishInitialization();
+
+                if (_context.HaveError)
+                {
+                    SendError();
+                    Close(true);
+                    return;
+                }
+
+                if (!_epl.BindContext(_context))
+                {
+                    SendError("Invalid host", 400);
```

Comment:
The http response strings are not localized on Windows, I dont think we need to localize them.